src/log.js

```diff
@@ -270,6 +270,30 @@ class Log extends GSet {
     return entry
   }
 
+  /*
+   * Creates a javscript iterator over log entries
+   *
+   * @param {Object} options
+   * @param {string} options.gte Beginning hash of the iterator
+   * @param {string} options.lte Ending hash of the iterator
+   * @param {number} options.number Number of entried to return to / from the gte / lte hash
+   * @returns {Symbol.Iterator} Iterator object containing log entries
+   *
+   */
+  iterator ({ gte = undefined, lte = undefined, amount = -1 } = {}) {
+    if(!gte && !lte) throw LogError.GteOrLteNotDefinedError()
```

Comment:
We probably don't need to force the `gte` or `lte`. If no options given, default should be to start iterating from the heads? What do you think?

We also want to have `gt` and `lt` in addition to `gte/lte` (ie. "iterate up until <gt/lt>, don't include the given hash").

Reply:
Sounds good to me!

Reply:
@haadcode should we throw an error in any event at all? like if gt and gte (or lt and lte) are passed simultaneously?